core-bundle/src/Image/Studio/ImageResult.php

```diff
@@ -142,6 +145,49 @@ public function getFilePath($absolute = false): string
         return $absolute ? $path : Path::makeRelative($path, $this->projectDir);
     }
 
+    /**
+     * Synchronously process images if they are deferred.
+     *
+     * This will make sure that the target files physically exist instead of
+     * being generated by the Contao\CoreBundle\Controller\ImagesController
+     * on first access.
+     */
+    public function createIfDeferred(): void
+    {
+        $picture = $this->getPicture();
+
+        $candidates = [];
+
+        foreach (array_merge([$picture->getImg()], $picture->getSources()) as $source) {
+            $candidates[] = $source['src'] ?? null;
+
+            foreach ($source['srcset'] ?? [] as $srcset) {
+                $candidates[] = $srcset[0] ?? null;
+            }
+        }
+
+        $deferredImages = array_filter(
+            $candidates,
+            static function ($image): bool {
+                return $image instanceof DeferredImageInterface;
+            }
+        );
+
+        if (empty($deferredImages)) {
+            return;
+        }
+
+        $resizer = $this->resizer();
```

Comment:
Shouldn't we check this first? If the resizer isn't available, we don't need to collect any `$deferredImages`.

Reply:
see https://github.com/contao/contao/pull/2368#discussion_r495817375